### PR TITLE
Do not hardcode http as plugin URL scheme for secure connections.

### PR DESCRIPTION
**Pull request by calavera**
Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

**Comment by abronan** [comment]
LGTM when Jenkins is happy

**Comment by runcom** [comment]
LGTM

File: pkg/plugins/client.go [hunk 1]
```diff
@@ -40,13 +40,19 @@ func NewClient(addr string, tlsConfig tlsconfig.Options) (*Client, error) {
...
+	}
+	return &Client{&http.Client{Transport: tr}, scheme, protoAndAddr[1]}, nil
 }
 
 // Client represents a plugin client.
 type Client struct {
-	http *http.Client // http client to use
-	addr string       // http address of the plugin
+	http   *http.Client // http client to use
+	scheme string       // scheme protocol of the plugin
```
Comment (tiborvass) [inline pkg/plugins/client.go]: Alternatively you could use a url.URL, it's up to you

Comment (calavera) [inline pkg/plugins/client.go]: yep, there are other approaches. The implications of parsing urls can be different, for instance with semantics between unix socket urls and http urls, so I decided to go with the simplest solution that fixes the problem.

Comment (tiborvass) [inline pkg/plugins/client.go]: ok makes sense

**Pull request merged**